code/Controllers/CMSMain.php

```diff
@@ -2309,11 +2277,11 @@ public function duplicate($request)
         return new HTTPResponse("CMSMain::duplicate() Bad ID: '$id'", 400);
     }
 
-    public function duplicatewithchildren($request)
+    public function duplicatewithchildren(HTTPRequest $request): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array